converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/FhirTransaction/RetryableFhirTransactionPipeline.cs

```diff
@@ -20,35 +20,52 @@ namespace Microsoft.Health.DicomCast.Core.Features.Worker.FhirTransaction
     public class RetryableFhirTransactionPipeline : IFhirTransactionPipeline
     {
         private const int MaxRetryCount = 3;
-        private static readonly TimeSpan SleepDuration = TimeSpan.FromMilliseconds(100);
 
         private readonly IFhirTransactionPipeline _fhirTransactionPipeline;
+        private readonly IExceptionStore _exceptionStore;
         private readonly IAsyncPolicy _retryPolicy;
 
-        public RetryableFhirTransactionPipeline(IFhirTransactionPipeline fhirTransactionPipeline)
+        public RetryableFhirTransactionPipeline(IFhirTransactionPipeline fhirTransactionPipeline, IExceptionStore exceptionStore)
             : this(
                   fhirTransactionPipeline,
-                  MaxRetryCount,
-                  retryCount => retryCount == 0 ? TimeSpan.Zero : SleepDuration)
+                  exceptionStore,
+                  MaxRetryCount)
         {
         }
 
         internal RetryableFhirTransactionPipeline(
             IFhirTransactionPipeline fhirTransactionPipeline,
-            int retryCount,
-            Func<int, TimeSpan> sleepDurationProvider)
+            IExceptionStore exceptionStore,
+            int maxRetryCount)
         {
             EnsureArg.IsNotNull(fhirTransactionPipeline, nameof(fhirTransactionPipeline));
+            EnsureArg.IsNotNull(exceptionStore, nameof(exceptionStore));
 
             _fhirTransactionPipeline = fhirTransactionPipeline;
+            _exceptionStore = exceptionStore;
             _retryPolicy = Policy
-                .Handle<ResourceConflictException>()
-                .Or<ServerTooBusyException>()
-                .WaitAndRetryAsync(retryCount, sleepDurationProvider);
+                .Handle<RetryableException>()
+                .Or<TaskCanceledException>()
+                .WaitAndRetryAsync(
+                    maxRetryCount,
+                    retryAttempt => TimeSpan.FromSeconds(Math.Pow(2, retryAttempt)),
```

Comment:
we should probably have a max on the exponential. doubling every retry can get quite long quite quickly

Reply:
Do you mean a max number of times to retry? Currently it is set for 3. Do you think it should be changed?